math/modular_inverse_simple.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file
+ * @brief simple implementation of adaption modular multiplicative inverse
+ * algorithm
+ *
+ * @details
+ * this algorithm calculate the modular inverse iteratively
+ *
+ * @ref
+ * https://en.wikipedia.org/wiki/Modular_multiplicative_inverse
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+
+/**
+ * @brief Function imod
+ *  Responsible for calculating the modular inverse from x % y
+ * @param x number
+ * @param y number
+ * @returns the modular inverse
+ */
+uint64_t imod(uint64_t x, uint64_t y) {  // positive integer expected
```

Comment:
What happens when a negative number is provided in the arguments - I don't think you've considered a fail safe here.